canvas_modules/common-canvas/src/common-properties/actions/action-factory.js

```diff
@@ -28,18 +28,49 @@ export default class ActionFactory {
 	constructor(controller) {
 		this.controller = controller;
 	}
+	// Retrieve icon for action button, with a new type "actionButtonIcon"
+	actionButtonIconCallback(buttonId, data) {
+		const buttonIconHandler = this.controller.getHandlers().buttonIconHandler;
+		const propertyId = { name: data.data.parameter_ref };
+		let icon;
+		if (buttonIconHandler) {
+			buttonIconHandler({
+				type: "actionButtonIcon",
+				propertyId: propertyId,
+				buttonId: buttonId,
+				data: data
+			}, (appIcon) => {
+				icon = appIcon;
+			}
+			);
+		}
+		return icon;
 
-
+	}
 	generateAction(key, action) {
 		if (action) {
 			if (action.actionType === ActionType.BUTTON) {
-				return (
-					<ButtonAction
-						key={"action." + key}
-						action={action}
-						controller={this.controller}
-					/>
-				);
+				let actionButton;
+				if (action.name === "iconButton") {
+					actionButton = (
+						<ButtonAction
```

Comment:
I don't understand why this extra code is needed.  All this can be done in ButtonAction without having to pass the extra parameters.